cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,362 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Testnet.Test.LedgerEvents.Gov.PredefinedAbstainDRep
+  ( hprop_check_predefined_abstain_drep
+  ) where
+
+import           Cardano.Api as Api
+import           Cardano.Api.Eon.ShelleyBasedEra (ShelleyLedgerEra)
+import           Cardano.Api.Error (displayError)
+
+import           Cardano.Ledger.Conway.Core (ppNOptL)
+import           Cardano.Ledger.Conway.Governance (ConwayGovState, cgsCurPParamsL)
+import           Cardano.Ledger.Core (EraPParams)
+import           Cardano.Ledger.Shelley.LedgerState (epochStateGovStateL, nesEpochStateL)
+import           Cardano.Testnet
+
+import           Prelude
+
+import           Control.Monad (void)
+import           Control.Monad.Catch (MonadCatch)
+import           Data.String (fromString)
+import qualified Data.Text as Text
+import           Data.Word (Word32)
+import           GHC.Stack (HasCallStack, callStack)
+import           Lens.Micro ((^.))
+import           System.FilePath ((</>))
+
+import           Testnet.Components.DReps (createCertificatePublicationTxBody, createVotingTxBody,
+                   generateVoteFiles, retrieveTransactionId, signTx, submitTx)
+import           Testnet.Components.Query (EpochStateView, findLargestUtxoForPaymentKey,
+                   getCurrentEpochNo, getEpochStateView, getGovState, getMinDRepDeposit,
+                   waitAndCheckNewEpochState)
+import           Testnet.Defaults (defaultDRepKeyPair, defaultDelegatorStakeKeyPair)
+import qualified Testnet.Process.Cli as P
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+import           Hedgehog
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+
+-- | This test creates a default testnet with three DReps and one stake holder delegated to each.
+-- We then do a proposal for an arbitrary parameter change (in this case to the
```

Comment:
You should mention voting thresholds here. Technically they could be set so that the proposal succeeds in this scenario.

Reply:
Hm, not in this case, cause we have more "no"s than "yes"es, so threshold would not help. But it would make the abstentions unnecessary in the second case, so I can say that.

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/2d3c9e6d9aa8bc82a5f0f474041f1ed1f8fdcd88